### PR TITLE
Homework week3

### PR DESCRIPTION
**Pull request by neveenatik**
Thanks for review

### CODE REVIEW

File: homework-submission/week3/src/App.js [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+  handleSubmit = () => {
+    const newItem = {
+      id: this.state.newId,
+      description: this.state.newDescription,
+      deadline: this.state.newDeadline,
+      done: false,
+      editable: false
+    };
+
+    let newState = this.state;
```
Comment (eidosam) [inline homework-submission/week3/src/App.js]: Use `const`

File: homework-submission/week3/src/App.js [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    newState.newDeadline = "";
+
+    localStorage.setItem("items", JSON.stringify(newState.items));
+    this.setState(newState);
+  };
+
+  handleDescriptionChange = (newDescription) => {
+    let newState = this.state;
+    newState.newDescription = newDescription;
+    localStorage.setItem("items", JSON.stringify(newState.items));
```
Comment (eidosam) [inline homework-submission/week3/src/App.js]: `items` is not changed, no need to update it in local-storage

File: homework-submission/week3/src/App.js [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+    newId:
+      localStorage.items !== undefined
+        ? JSON.parse(localStorage.items).length + 1
+        : TodoItems.length + 1,
+    newDescription: "",
+    newDeadline: ""
+  };
+
+  handleChecked = id => {
+    let newState = this.state;
```
Comment (eidosam) [inline homework-submission/week3/src/App.js]: Use `const` when you don't re-assign

File: homework-submission/week3/src/App.js [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+    newState.items.push(newItem);
+    newState.newDescription = "";
+    newState.newDeadline = "";
+
+    localStorage.setItem("items", JSON.stringify(newState.items));
+    this.setState(newState);
+  };
+
+  handleDescriptionChange = (newDescription) => {
+    let newState = this.state;
```
Comment (eidosam) [inline homework-submission/week3/src/App.js]: Use `const`

File: homework-submission/week3/src/App.js [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+    newState.newDescription = newDescription;
+    localStorage.setItem("items", JSON.stringify(newState.items));
+    this.setState(newState);
+  };
+
+  handleDeadLineChange = (value) => {
+    let newState = this.state;
```
Comment (eidosam) [inline homework-submission/week3/src/App.js]: `const` instead of `let`

File: homework-submission/week3/src/App.js [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+    newState.newDeadline = value;
+    localStorage.setItem("items", JSON.stringify(newState.items));
```
Comment (eidosam) [inline homework-submission/week3/src/App.js]: Same note about `items` not been updated.

File: homework-submission/week3/src/App.js [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+    this.setState(newState);
+  };
+
+  handleRemove = (id) => {
+    let newState = this.state;
```
Comment (eidosam) [inline homework-submission/week3/src/App.js]: `const`

File: homework-submission/week3/src/App.js [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+
+  handleEdit = (id) => {
+    let newState = this.state;
+    id = parseInt(id, 10);
+    const indexOfItem = newState.items.findIndex(item => item.id === id);
+    if (indexOfItem >= 0) {
+      newState.items[indexOfItem].editable === false
+        ? (newState.items[indexOfItem].editable = true)
+        : (newState.items[indexOfItem].editable = true);
+    }
```
Comment (eidosam) [inline homework-submission/week3/src/App.js]: Use `Logical NOT (!)`

File: homework-submission/week3/src/App.js [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+    localStorage.setItem("items", JSON.stringify(newState.items));
+    this.setState({ items: newState.items });
+  };
+
+  handleUpdate = (id) => {
+    
+    id = parseInt(id, 10);
+    let newState = this.state;
```
Comment (eidosam) [inline homework-submission/week3/src/App.js]: `const`

File: homework-submission/week3/src/App.js [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+      newState.newDescription = "";
+      newState.newDeadline = "";
+    }
+
+    localStorage.setItem("items", JSON.stringify(newState.items));
+    this.setState(newState);
+  };
+
+  handleCancel = id => {
+    let newState = this.state;
```
Comment (eidosam) [inline homework-submission/week3/src/App.js]: `const`

File: homework-submission/week3/src/App.js [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+    id = parseInt(id, 10);
+    const indexOfItem = newState.items.findIndex(item => item.id === id);
+    if (indexOfItem >= 0) {
+      newState.items[indexOfItem].editable === true
+        ? (newState.items[indexOfItem].editable = false)
+        : (newState.items[indexOfItem].editable = false);
```
Comment (eidosam) [inline homework-submission/week3/src/App.js]: `Logical NOT (!)`

File: homework-submission/week3/src/App.js [hunk 12]
```diff
@@ -0,0 +1,215 @@
...
+    localStorage.setItem("newDeadline", JSON.stringify(newState.newDeadline));
+    this.setState({ newDescription: newState.newDescription, newDeadline: newState.newDeadline });
+  };
+
+  componentDidMount() {
+    let itemsCopy = localStorage.items;
+
+    if (itemsCopy !== undefined) {
+      this.setState({ items: JSON.parse(itemsCopy) });
+    }
```
Comment (eidosam) [inline homework-submission/week3/src/App.js]: Local-Storage is already checked in the Constructor, no need to do it here

**Comment by neveenatik** [comment]
Hi @eidosam 
Thanks for your review, I appreciate all your feedback . I have made the needed changes.

**Pull request closed without merging**